### DISCUSSION TITLE
User not automatically logging in after email confirmation

### DISCUSSION THREAD

**Discussion by Mr-Pepe** [supabase/supabase#27435] [Questions] [answered]
I am building a Flutter app and have registered a listener on the `client.auth.onAuthStateChange` stream. 
A simple dialog allows registering an account. They user fills in an email and a password, I call 
```dart 
client.auth.signUp(
  email: email,
  password: password,
  emailRedirectTo: "mydomain.com/registration-successful",
)
```
and the user receives an email. The dialog tells the user to check their email. The user clicks the link in the email and gets redirected to a web page that says "Well done, you can now go back to the app and enjoy all the features".

However, it seems like clicking the link did not trigger a login. So the listener to the auth changes does not get notified and the popup stays in place and tells the user to check their email. I would like the popup to disappear once the user has confirmed their email though.

What is the expected behavior? Does the user need to explicitly log in with their email and password after confirming their email?

**Comment by j4w8n** [comment, marked as answer]
They should be automatically logged in after clicking the confirmation link.

However, I'm not familiar with the dialog process.

**Reply by Mr-Pepe** [reply]
I was under the impression that simply clicking the confirmation link was enough to sign the user in. However, it seems that the redirect link can not link to a generic web page, but must link back to the app itself and then the Supabase instance does something with the `code` query parameter. I guess I haven't fully understood how this whole thing works yet.

**Reply by j4w8n** [reply]
Yes, that's more technically accurate, but it should be so quick that it appears seamless.

**Reply by Mr-Pepe** [reply]
And it seems that `client.auth.onAuthStateChange` does not work across tabs. 

Let's say somewhere deep down in the app, there is a feature that requires an account. I would like for my users to be able to click a button, get a pop-up, sign up, get the email, click the link, get redirected to some page that says "well done, now keep working in your original tab", switch back to the original tab and now have all features enabled. That doesn't seem possible right now though.

**Reply by silentworks** [reply]
@Mr-Pepe this isn't something specific to Supabase but more a development practice. You would just attach a query param to the redirect url with the current page the user is on and redirect them when they return to the site from the link in the email. There is example code of this in the Supabase docs where the `next` querystring is used to store the user's current location. I'm not sure how you pass path to tabs in Flutter so can't comment on the implementation of this.
